x-pack/plugins/reporting/server/export_types/csv_searchsource/generate_csv/generate_csv.ts

```diff
@@ -135,27 +136,37 @@ export class CsvGenerator {
     };
   }
 
-  // use fields/fieldsFromSource from the searchSource to get the ordering of columns
-  // otherwise use the table columns as they are
-  private getFields(searchSource: ISearchSource, table: Datatable): string[] {
-    const fieldValues: Record<string, string | boolean | SearchFieldValue[] | undefined> = {
-      fields: searchSource.getField('fields'),
-      fieldsFromSource: searchSource.getField('fieldsFromSource'),
-    };
-    const fieldSource = fieldValues.fieldsFromSource ? 'fieldsFromSource' : 'fields';
-    this.logger.debug(`Getting search source fields from: '${fieldSource}'`);
-
-    const fields = fieldValues[fieldSource];
-    // Check if field name values are string[] and if the fields are user-defined
-    if (isPlainStringArray(fields)) {
-      return fields;
+  private getColumns(searchSource: ISearchSource, table: Datatable) {
+    if (this._columns != null) {
+      return this._columns;
     }
 
-    // Default to using the table column IDs as the fields
-    const columnIds = table.columns.map((c) => c.id);
-    // Fields in the API response don't come sorted - they need to be sorted client-side
-    columnIds.sort();
-    return columnIds;
+    // if columns is not provided in job params,
+    // default to use fields/fieldsFromSource from the searchSource to get the ordering of columns
+    const getFromSearchSource = (): string[] => {
+      const fieldValues: Record<string, string | boolean | SearchFieldValue[] | undefined> = {
```

Comment:
What do you think about reusing the `SearchSourceFields` interface instead of record here? That seems like what we need.